diff -rupN RoBoIO-src-v1.6-gcc/libsrc/rcservo.cpp RoBoIO-src-v1.6-gcc.mine/libsrc/rcservo.cpp
--- RoBoIO-src-v1.6-gcc/libsrc/rcservo.cpp	2010-05-05 22:13:30.000000000 +0900
+++ RoBoIO-src-v1.6-gcc.mine/libsrc/rcservo.cpp	2010-06-30 00:46:23.000000000 +0900
@@ -704,7 +704,7 @@ _RB_INLINE void rcservo_ReadPositions_Pl
 	    if (width[i] == 0xffffffffL)
 		{
 			width[i] = 0L;
-			RCSERVO_validChannels = RCSERVO_validChannels & (~(1L<<i));
+			//RCSERVO_validChannels = RCSERVO_validChannels & (~(1L<<i));
 
 			errmsg("WARNING: %s() fails to read channel %d!\n", __FUNCTION__, i);
 		}
diff -rupN RoBoIO-src-v1.6-gcc/Make.LINUX RoBoIO-src-v1.6-gcc.mine/Make.LINUX
--- RoBoIO-src-v1.6-gcc/Make.LINUX	2009-09-18 19:33:04.000000000 +0900
+++ RoBoIO-src-v1.6-gcc.mine/Make.LINUX	2010-07-01 18:13:12.000000000 +0900
@@ -59,5 +59,5 @@ userio.o: example/userio.cpp example/use
 
 .PHONY : clean
 clean :
-	-rm RoBoRC.exe libRBIO.a $(RBDEMO_OBJECTS) $(ROBOIO_OBJECTS)
+	-rm -f RoBoRC.exe libRBIO.a $(RBDEMO_OBJECTS) $(ROBOIO_OBJECTS)
 
